kong/plugins/azure-functions/handler.lua

```diff
@@ -115,22 +148,30 @@ function azure:access(config)
     return responses.send_HTTP_INTERNAL_SERVER_ERROR(err)
   end
 
-  -- prepare response for downstream
-  for k, v in pairs(res.headers) do
-    response_headers[k] = v
-  end
-
-  response_headers.Server = SERVER
-  ngx.status = res.status
-  local response_body = res:read_body()
-  ngx_print(response_body)
+  local response_headers = res.headers
+  local response_status = res.status
+  local response_content = res:read_body()
 
   ok, err = client:set_keepalive(config.keepalive)
   if not ok then
     ngx_log(ngx.ERR, "[azure-functions] could not keepalive connection: ", err)
   end
 
-  return ngx_exit(res.status)
+  local ctx = ngx.ctx
+  if ctx.delay_response and not ctx.delayed_response then
+    ctx.delayed_response = {
+      status_code               = response_status,
+      content                   = response_content,
+      headers                   = response_headers,
+    }
+
+    ctx.delayed_response_callback = flush
+
+    coroutine.yield()
```

Comment:
This shouldn't be needed, the `return` from the line below should already suffice to resume the plugins iterator. Unless there is another reason that I am not seeing, we should not confuse plugin authors into thinking this is necessary.